### PR TITLE
support ES document type

### PR DESCRIPTION
**Pull request by code4value**

### CODE REVIEW

File: es2csv.py [hunk 1]
```diff
@@ -256,6 +259,7 @@ def main():
     p.add_argument('-q', '--query', dest='query', type=str, required=True, help='Query string in Lucene syntax.')
     p.add_argument('-u', '--url', dest='url', default='http://localhost:9200', type=str, help='Elasticsearch host URL. Default is %(default)s.')
     p.add_argument('-i', '--index-prefixes', dest='index_prefixes', default=['logstash-*'], type=str, nargs='+', metavar='INDEX', help='Index name prefix(es). Default is %(default)s.')
+    p.add_argument('-D', '--doc_type', dest='doc_type', default=['logstash-*'], type=str, nargs='+', metavar='DOC_TYPE', help='Document type. Default is %(default)s.')
```
Comment (taraslayshchuk) [inline es2csv.py]: Please add example off usage to readme.md.

File: README.rst [hunk 2]
```diff
@@ -38,6 +38,7 @@ Usage
   -o, --output_file FILE                   CSV file location.                           [required]
   -u, --url URL                            Elasticsearch host URL. Default is http://localhost:9200.
   -i, --index-prefixes INDEX [INDEX ...]   Index name prefix(es). Default is ['logstash-*'].
+  -D, --doc_type doc_type [doc_type ...]   Document type. Default is ['logstash-*'].
```
Comment (taraslayshchuk) [inline README.rst]: `doc_type – A comma-separated list of document types to search; leave empty to perform the operation on all types`[[link](http://elasticsearch-py.readthedocs.io/en/2.3.0/api.html#elasticsearch.Elasticsearch.search)]
I am not sure that we need default value for this param.
Please change doc_type to upper case.

File: es2csv.py [hunk 3]
```diff
@@ -99,6 +99,9 @@ def next_scroll(scroll_id):
             size=self.scroll_size,
             terminate_after=self.opts.max_results
         )
+        if self.opts.doc_type:
+            search_args['doc_type']=self.opts.doc_type
```
Comment (taraslayshchuk) [inline es2csv.py]: Please format this code according to [PEP 8](https://www.python.org/dev/peps/pep-0008/).

**Comment by code4value** [comment]
thanks for feedback @taraslayshchuk. Code updated accordingly. Since multiple values supported for doc types, option renamed to doc_types.

**Comment by taraslayshchuk** [comment]
Thanks, according to renaming could you please correct the [readme] (https://github.com/taraslayshchuk/es2csv/pull/13/commits/244cc771205d584dff3034158c82af26e34bc85f#diff-88b99bb28683bd5b7e3a204826ead112R41) - `--doc_type -> --doc_types` ?

File: README.rst [hunk 4]
```diff
@@ -135,6 +136,12 @@ Collecting all data on all indices
 
   $ es2csv -i _all -q '*' -o database.csv
   
+Specifying document type
+
+.. code-block:: bash
+
+  $ es2csv -i _all -D log -q '*' -o database.csv
```
Comment (taraslayshchuk) [inline README.rst]: And please move `-D` arg in front of `-i`.

**Comment by taraslayshchuk** [comment]
And there is conflicts now, @code4value could you please resolve it?

**Comment by code4value** [comment]
readme updated and conflict resolved

File: README.rst [hunk 5]
```diff
@@ -29,7 +29,7 @@ Usage
 -----
 .. code-block:: bash
 
- $ es2csv [-h] -q QUERY [-u URL] [-a AUTH] [-i INDEX [INDEX ...]]
+ $ es2csv [-h] -q QUERY [-u URL] [-a AUTH] [-i INDEX [INDEX ...]] [-D DOC_TYPE [DOC_TYPE ...]]
```
Comment (taraslayshchuk) [inline README.rst]: $ es2csv.py [-h] -q QUERY [-u URL] [-a AUTH] [-i INDEX [INDEX ...]]
                 [-D DOC_TYPE [DOC_TYPE ...]] [-t TAGS [TAGS ...]] -o FILE
                 [-f FIELDS [FIELDS ...]] [-d DELIMITER] [-m INTEGER] [-k]
                 [-r] [-e] [-v] [--debug]

File: README.rst [hunk 6]
```diff
@@ -39,6 +39,7 @@ Usage
   -u, --url URL                            Elasticsearch host URL. Default is http://localhost:9200.
   -a, --auth                               Elasticsearch basic authentication in the form of username:password.
   -i, --index-prefixes INDEX [INDEX ...]   Index name prefix(es). Default is ['logstash-*'].
+  -D, --doc_types DOC_TYPE [DOC_TYPE ...]   Document type(s).
```
Comment (taraslayshchuk) [inline README.rst]: -D, --doc_types DOC_TYPE [DOC_TYPE ...]  Document type(s).

**Comment by taraslayshchuk** [comment]
Great job, thanks for adding this feature!

**Pull request merged**